data_steward/cdr_cleaner/cleaning_rules/ppi_branching.py

```diff
@@ -0,0 +1,331 @@
+        rules_dataframe = self.create_rules_dataframe()
+        job_config.write_disposition = bigquery.WriteDisposition.WRITE_TRUNCATE
+        job = client.load_table_from_dataframe(rules_dataframe,
+                                               destination=self.lookup_table,
+                                               job_config=job_config)
+        return job.result()
+
+    def backup_rows_to_drop_ddl(self) -> str:
+        """
+        Get a DDL statement which loads a backup table with rows to be dropped
+
+        :return: the DDL statement
+        """
+        observation_schema = bq.get_table_schema(OBSERVATION)
+        query = BACKUP_ROWS_QUERY.render(lookup_table=self.lookup_table,
+                                         src_table=self.observation_table)
+        return bq.get_table_ddl(dataset_id=self.backup_table.dataset_id,
+                                table_id=self.backup_table.table_id,
+                                schema=observation_schema,
+                                as_query=query)
+
+    def stage_cleaned_table_ddl(self) -> str:
+        """
+        Get a DDL statement which stages cleaned table
+
+        Note: This avoids potential partitioning mismatch error
+              when directly overwriting observation table
+
+        :return: the DDL statement
+        """
+        observation_schema = bq.get_table_schema(OBSERVATION)
+        query = CLEANED_ROWS_QUERY.render(src=self.observation_table,
+                                          backup=self.backup_table)
+        return bq.get_table_ddl(dataset_id=self.stage_table.dataset_id,
+                                table_id=self.stage_table.table_id,
+                                schema=observation_schema,
+                                as_query=query)
+
+    def drop_observation_ddl(self) -> str:
+        """
+        Get a DDL statement which drops the observation table
+
+        :return: the DDL statement
+        """
+        table = self.observation_table
+        return f'DROP TABLE `{table.project}.{table.dataset_id}.{table.table_id}`'
+
+    def stage_to_target_ddl(self) -> str:
+        """
+        Get a DDL statement which drops and creates the observation
+        table with rows from stage
+
+        :return: the DDL statement
+        """
+        observation_schema = bq.get_table_schema(OBSERVATION)
+        stage = self.stage_table
+        query = f'''SELECT * FROM `{stage.project}.{stage.dataset_id}.{stage.table_id}`'''
+        return bq.get_table_ddl(self.observation_table.dataset_id,
+                                schema=observation_schema,
+                                table_id=self.observation_table.table_id,
+                                as_query=query)
+
+    def cleaning_script(self) -> str:
+        """
+        Get script which cleans the observation table
+
+        :return: the SQL script
+        """
+        script = f"""
+        {self.backup_rows_to_drop_ddl()};
```

Comment:
The creation of `_ppi_branching_rules_lookup` needs to be added here (maybe using CREATE IF NOT EXISTS?)

Reply:
So _I did_ want to make loading the lookup a part of `script`, but I don't have an easy way to load `*.csv` files into a table while adding a `rule_source` column, all condensed in a single DDL statement. :disappointed:  Instead, loading the lookup happens in `setup_rule()` which this assumes gets called prior, but come to think of it, [clean_cdr does NOT do this](https://github.com/all-of-us/curation/blob/develop/data_steward/cdr_cleaner/clean_cdr.py#L490-L494) yet. In the meantime, shall I do something similar to what we are doing for `UnitNormalization`?

Reply:
This makes sense. Thanks for bringing that up, Mark. I think it's fine as is. I am working on running the `setup_rule()` via `clean_dataset.py` before the cleaning rule runs, which should handle this case.

Reply:
Ah ok thanks @nishanthpp93. I will leave out calling `setup_rule()` and let your upcoming fix address this then.